tests/phpunit/api/v3/PaymentTest.php

```diff
@@ -0,0 +1,311 @@
+
+    //Create partial payment
+    $params = array(
+      'contribution_id' => $contribution['id'],
+      'total_amount' => 150,
+    );
+    $payment = $this->callAPIAndDocument('payment', 'create', $params, __FUNCTION__, __FILE__);
+
+    $this->assertEquals($payment['values'][$payment['id']]['from_financial_account_id'], 7);
+    $this->assertEquals($payment['values'][$payment['id']]['to_financial_account_id'], 6);
+    $this->assertEquals($payment['values'][$payment['id']]['total_amount'], 150);
+    $this->assertEquals($payment['values'][$payment['id']]['status_id'], 1);
+    $this->assertEquals($payment['values'][$payment['id']]['is_payment'], 1);
+
+    // Check entity financial trxn created properly
+    $params = array(
+      'entity_id' => $contribution['id'],
+      'entity_table' => 'civicrm_contribution',
+      'financial_trxn_id' => $payment['id'],
+    );
+
+    $eft = $this->callAPISuccess('EntityFinancialTrxn', 'get', $params);
+
+    $this->assertEquals($eft['values'][$eft['id']]['amount'], 150);
+
+    // Now create payment to complete total amount of contribution
+    $params = array(
+      'contribution_id' => $contribution['id'],
+      'total_amount' => 50,
+    );
+    $payment = $this->callAPIAndDocument('payment', 'create', $params, __FUNCTION__, __FILE__);
+
+    $this->assertEquals($payment['values'][$payment['id']]['from_financial_account_id'], 7);
+    $this->assertEquals($payment['values'][$payment['id']]['to_financial_account_id'], 6);
+    $this->assertEquals($payment['values'][$payment['id']]['total_amount'], 50);
+    $this->assertEquals($payment['values'][$payment['id']]['status_id'], 1);
+    $this->assertEquals($payment['values'][$payment['id']]['is_payment'], 1);
+
+    // Check contribution for completed status
+    $contribution = $this->callAPISuccess('contribution', 'get', array('id' => $contribution['id']));
+
+    $this->assertEquals($contribution['values'][$contribution['id']]['contribution_status'], 'Completed');
+    $this->assertEquals($contribution['values'][$contribution['id']]['total_amount'], 200.00);
+
+    $this->callAPISuccess('Contribution', 'Delete', array(
+      'id' => $contribution['id'],
+    ));
+  }
+
+  public function testCreatePaymentLineItems() {
+
+    // Create priceset & price fields
+    $this->createPriceSet();
+
+    $params = array(
+      'contact_id' => $this->_individualId,
+      'receive_date' => '20120511',
+      'total_amount' => 200.00,
+      'financial_type_id' => $this->_financialTypeId,
+      'payment_instrument_id' => 1,
+      'contribution_status_id' => 8,
+    );
+
+    $contribution = $this->callAPISuccess('contribution', 'create', $params); //Create partially paid contribution
+
+    //Create partial payment
+    $params = array(
+      'contribution_id' => $contribution['id'],
+      'total_amount' => 150,
+    );
+    $payment = $this->callAPIAndDocument('payment', 'create', $params, __FUNCTION__, __FILE__);
```

Comment:
A minor point - the callAPIAndDocument is called multiple times with the same parameters. This will result in the example being overwritten multiple times. It's best just to call it for the version you want to create an example for and use callAPISuccess for the others.

Note the default buildkit install puts a setting in the civicrm_settings.php file that blocks example generation - to generate an example you need to edit that, run the tests and commit the example file, while will then show in api explorer.

I don't think this should be a blocker on merging this PR but if you have to touch it again you might like to think about committing an example & changing the other calls to callAPISuccess